AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m

```diff
@@ -369,9 +384,21 @@ - (void)trackPage:(NSString *)pageName withProperties:(NSDictionary<NSString *,
 }
 
 - (void)sendLog:(id<MSLog>)log flags:(MSFlags)flags {
+  [(flags & MSFlagsCritical) != 0 ? self.criticalChannelUnit : self.channelUnit enqueueItem:log flags:flags];
+}
 
-  // Send log to log manager.
-  [self.channelUnit enqueueItem:log flags:flags];
+- (void)setTransmissionInterval:(NSUInteger)interval {
+  if (self.started) {
+    MSLogError([MSAnalytics logTag], @"The transmission interval should be set before the MSAnalytics service is started.");
+    return;
+  }
+  if (interval > kMSFlushIntervalMaximum || interval < kMSFlushIntervalMinimum) {
+    MSLogError([MSAnalytics logTag], @"The transmission interval is not valid, it should be between %u seconds and %u seconds (%u days).",
+               (unsigned int)kMSFlushIntervalMinimum, (unsigned int)kMSFlushIntervalMaximum,
+               (unsigned int)(kMSFlushIntervalMaximum / 86400));
+    return;
+  }
+  self.flushInterval = interval;
```

Comment:
We probably want to log something to the console when the interval is set successfully.